include/zephyr/arch/arm/aarch32/cortex_a_r/sys_io.h

```diff
@@ -26,7 +26,9 @@ extern "C" {
 
 static ALWAYS_INLINE uint8_t sys_read8(mem_addr_t addr)
 {
-	uint8_t val = *(volatile uint8_t *)addr;
+	uint8_t val;
+
+	__asm__ volatile("ldrb %r0, [%1]" : "=r" (val) : "r" (addr));
```

Comment:
what does `r` in `%r0` signify? why should it not be just `%0`?

Reply:
Sorry, this is my mistake. I was assuming that %r would mean a register.
Looking at https://developer.arm.com/documentation/100067/0611/armclang-Inline-Assembler/Inline-assembly-template-modifiers/Template-modifiers-for-AArch32-state?lang=en , it has no meaning. I will fix this in v2.